core/validation/failures.py

```diff
@@ -197,6 +197,7 @@ class WrongTypeFailure(ValidationFailure):
     column: str
     expected_type: str
     actual_type: str
+    index: list
```

Comment:
maybe not one for this PR but do we expect all `ValidationFailures` to have indexes? I suppose possibly all non-pre-transformation-failures? I think we should look to move these up the inheritance hierarchy if so

Reply:
We do, and we don't. We still haven't managed to retrieve indexes for footfall outcomes, and that's going to have to be accounted for in the way the failure's to_message() method works. Otherwise, for all of the errors we expect (i.e. the ones we're writing error messages for) we can retain and use the index.